### DISCUSSION TITLE
devices priority?

### DISCUSSION THREAD

**Discussion by habub** [merbanan/rtl_433#2709] [Q&A] [answered]
Hi,
What is the **r_dev->priority** parameter? I couldn't find it in the devices files. Is it used? Where is it defined?
How does it impact the _run_ook_demods(list_t* r_devs, pulse_data_t* pulse_data)_ function?

Thanks

**Comment by zuckschwerdt** [comment, marked as answer]
Priority let's a decoder run later and only if no previous events were produced.
See #1895

**Reply by habub** [reply]
"events" meaning successful decoding?
Currenty all devices are same priority (0), right?

**Reply by zuckschwerdt** [reply]
Yes, successful decoding is an event.
Some decoders are bumped down with `.priority = 10`, there are notes on each as to why.
